java/client/src/main/java/com/youtube/vitess/client/Proto.java

```diff
@@ -203,6 +210,26 @@ public EntityId apply(Map.Entry<byte[], ?> entry) {
         // Float, Double
         this.type = Query.Type.FLOAT64;
         this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Date) {
+        // Date
+        this.type = Query.Type.DATE;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Time) {
+        // Time
+        this.type = Query.Type.TIME;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Timestamp) {
+        // DateTime, TimeStamp
+        this.type = Query.Type.TIMESTAMP;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Boolean ) {
+        // Boolean
+        this.type = Query.Type.BIT;
+        this.value = ByteString.copyFromUtf8(((boolean)value) ? "1" : "0");
+      } else if (value instanceof BigDecimal) {
+        // BigDecimal
+        this.type = Query.Type.FLOAT64;
```

Comment:
In MySQL you can have a scale of atmost 30.
So, If one gives value of BigDecimal as 0.23 and I store is as toPlainString then the value saved in database will be "0.230000000000000009992007221626" (Scale 30) and not "0.23" so on retrieval, they will get wrong value